src/lightcurvelynx/simulate.py

```diff
@@ -221,10 +333,100 @@ def simulate_lightcurves(
+    pool=None,
+    batch_size=1_000,
+):
+    """Generate a number of simulations of the given model and information
+    from one or more surveys.
+
+    Parameters
+    ----------
+    model : BasePhysicalModel
+        The model to draw from. This may have its own parameters which
+        will be randomly sampled with each draw.
+    num_samples : int
+        The number of samples.
+    obstable : ObsTable or List of ObsTable
+        The ObsTable(s) from which to extract information for the samples.
+    passbands : PassbandGroup or List of PassbandGroup
+        The passbands to use for generating the bandfluxes.
+    apply_obs_mask: boolean
+        If True, apply obs_mask to filter interesting indices/times.
+    time_window_offset : tuple(float, float), optional
+        A tuple specifying the time window offset (before, after) t0 in days.
+        This is used to filter the observations to only those within the specified
+        time window (t0 - before, t0 + after). If None or the model does not have a
+        t0 specified, no time window is applied.
+    obstable_save_cols : list of str, optional
+        A list of ObsTable columns to be saved as part of the results. This is used
+        to save context information about how the light curves were generated. If the column
+        is missing from one of the ObsTables, a null value such as None or NaN is used.
+        If None, no additional columns are saved.
+    param_cols : list of str, optional
+        A list of the model's parameter columns to be saved as separate columns in
+        the results (instead of just the full dictionary of parameters). These
+        must be specified as strings in the node_name.param_name format.
+        If None, no additional columns are saved.
+    rng : numpy.random._generator.Generator, optional
+        A given numpy random number generator to use for this computation. If not
+        provided, the function uses the node's random number generator.
+    generate_citations : bool, optional
+        If True, generate citations for the simulation and output them.
+        Default is False.
+    pool : multiprocessing.Pool, optional
+        An optional multiprocessing pool to use for parallel processing. If None,
+        the function runs in serial.
+    batch_size : int, optional
+        The number of samples to process in each batch when using multiprocessing.
+        Default is 1000.
+
+    Returns
+    -------
+    lightcurves : nested_pandas.NestedFrame
+        A NestedFrame with a row for each object.
+    """
+    simulation_info = SimulationInfo(
+        model=model,
+        num_samples=num_samples,
+        obstable=obstable,
+        passbands=passbands,
+        rng=rng,
+        apply_obs_mask=apply_obs_mask,
+        time_window_offset=time_window_offset,
+        obstable_save_cols=obstable_save_cols,
+        param_cols=param_cols,
+    )
+
+    if pool is not None and batch_size < num_samples:
```

Comment:
I'd always use the provided pool, especially if it represents some distributed cluster.